bindings/gumjs/gumcmodule.c

```diff
@@ -396,51 +479,62 @@ gum_cmodule_new (const gchar * source,
 void
 gum_cmodule_free (GumCModule * cmodule)
 {
+  if (cmodule == NULL)
+    return;
+
+  g_object_unref (cmodule);
 }
 
 const GumMemoryRange *
 gum_cmodule_get_range (GumCModule * self)
 {
-  g_assert_not_reached ();
-  return NULL;
+  g_assert (self != NULL);
+
+  return GUM_CMODULE_GET_CLASS (self)->get_range (self);
 }
 
 void
 gum_cmodule_add_symbol (GumCModule * self,
                         const gchar * name,
                         gconstpointer value)
 {
-  g_assert_not_reached ();
+  g_assert (self != NULL);
+
+  GUM_CMODULE_GET_CLASS (self)->add_symbol (self, name, value);
 }
 
 gboolean
 gum_cmodule_link (GumCModule * self,
                   GError ** error)
 {
-  g_assert_not_reached ();
-  return FALSE;
+  g_assert (self != NULL);
+
+  return GUM_CMODULE_GET_CLASS (self)->link (self, error);
 }
 
 void
 gum_cmodule_enumerate_symbols (GumCModule * self,
                                GumFoundCSymbolFunc func,
                                gpointer user_data)
 {
-  g_assert_not_reached ();
+  g_assert (self != NULL);
+
+  GUM_CMODULE_GET_CLASS (self)->enumerate_symbols (self, func, user_data);
 }
 
 gpointer
 gum_cmodule_find_symbol_by_name (GumCModule * self,
                                  const gchar * name)
 {
-  g_assert_not_reached ();
-  return NULL;
+  g_assert (self != NULL);
+
+  return GUM_CMODULE_GET_CLASS (self)->find_symbol_by_name (self, name);
 }
 
 void
 gum_cmodule_drop_metadata (GumCModule * self)
 {
-  g_assert_not_reached ();
-}
+  g_assert (self != NULL);
```

Comment:
I think we can omit assertions like these as passing `NULL` will result in a crash on the next line.